dvc/lock.py

```diff
@@ -101,44 +151,9 @@ def __del__(self):
                 pass
 
 
-else:
-    import zc.lockfile
-
-    class Lock(object):
-        """Class for dvc repo lock.
-
-        Uses zc.lockfile as backend.
-        """
-
-        def __init__(self, lockfile, tmp_dir=None):
-            self.lockfile = lockfile
-            self._lock = None
-
-        @property
-        def files(self):
-            return [self.lockfile]
-
-        def _do_lock(self):
-            try:
-                self._lock = zc.lockfile.LockFile(self.lockfile)
-            except zc.lockfile.LockError:
-                raise LockError(FAILED_TO_LOCK_MESSAGE)
-
-        def lock(self):
-            try:
-                self._do_lock()
-                return
-            except LockError:
-                time.sleep(DEFAULT_TIMEOUT)
-
-            self._do_lock()
-
-        def unlock(self):
-            self._lock.close()
-            self._lock = None
-
-        def __enter__(self):
-            self.lock()
+def make_lock(lockfile, tmp_dir=None, friendly=False, hardlink_lock=False):
+    if hardlink_lock and is_py2:
+        raise DvcException("Hardlink locks are not supported on Python <3.5.")
```

Comment:
are they set in a config file? Could we put a hint where to look for the setting?

Reply:
@shcheklein , this is an opt-in feature, I don't think there's a necessity to clear this out)

Reply:
one user opt-ins and someone who downloads the project has an error? is it a possible scenario?

Reply:
could be @shcheklein; by the way, it reminded me that I think this feature shouldn't be on the repo config, as it is a machine-specific work-around / limitation.

But I was about to put it on another issue about `config` itself :sweat_smile:

Reply:
Yes, added a link to docs.